alf/environments/gym_wrappers.py

```diff
@@ -54,6 +57,23 @@ def _traverse_transform(space, levels):
         space=observation_space, levels=field.split('.') if field else [])
 
 
+def need_channel_transpose(shape):
+    """Determine if the input shape needs a transpose to make the corresponding array/tensor
+        channel_first.
+
+    Args:
+        shape (int tuple): shape of the inquiry array / tensor
+
+    Returns:
+        A boolean: the inquiry array / tensor needs tranpose (True) or not (False)
+    """
+
+    if len(shape) == 3 and shape[2] < shape[0] and shape[2] < shape[1]:
```

Comment:
this check is error prone?  there could be more channels than width or height?
It's probably best to leave this as a configuration that's controlled by gin?

Reply:
Yes this is problem that @hnyu  and I have been discussing for some time. Image inputs could happen to be a field of a Dict or just directly the observation, but given most image formats do not have more than 4 channels, this was a workaround. Do we want to make wrappers also gin configurable?

Reply:
> this check is error prone? there could be more channels than width or height?
> It's probably best to leave this as a configuration that's controlled by gin?

The chance of having channels > max(width, height) is extremely low. So far I haven't seen any input data like that. Note that we are talking about raw input data from the environment, not some intermediate features output from a network.